doc/side-by-side-CTL-and-Plutus-comparison.md

```diff
@@ -0,0 +1,323 @@
+# Side by side comparison of CTL and Plutus off-chain contracts
+
+We are going to compare two different off-chain contracts 
+both in Plutus and CTL. 
+
+For this discussion, we need to go through an overview of 
+both of them.
+
+
+## About `Contract` in CTL and Plutus
+
+The purpose of `Contract` in CTL is to bring the capabilities of 
+`QueryMExtended` to the public API. 
+
+The current definition of `Contract` in CTL is :
+
+```PureScript
+type QueryConfig (r :: Row Type) =
+  { some_internal_parameters 
+  | r
+  }
+
+type DefaultQueryConfig = QueryConfig ()
+
+type QueryM (a :: Type) = ReaderT DefaultQueryConfig (LoggerT Aff) a
+
+type QueryMExtended (r :: Row Type) (a :: Type) =
+  ReaderT 
+    (QueryConfig r)
+    (LoggerT Aff)
+    a
+
+newtype Contract (r :: Row Type) (a :: Type) = Contract (QueryMExtended r a)
+```
+
+
+In CTL we have a general environment type `forall r . QueryConfig r`, it stores
+the needed parameters to connect to an `Ogmios` server  and
+uses the PureScript native row polymorphism to make it extensible for both 
+CTL developers and users.
```

Comment:
Let's be specific that it is the `r :: Row Type` that allows this extensibility as not everyone will be familiar with this.

Reply:
I'm not sure how to add this to the document without going into a discussion about `Haskell` vs `PS` type systems instead of `Plutus` vs `CTL`, so I just add a Wikipedia link and a link to [other source ](https://hgiasac.github.io/posts/2018-11-18-Record-Row-Type-and-Row-Polymorphism.html)